test/configuration_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "../src/configuration.h"
+#include "gtest/gtest.h"
+
+namespace google {
+
+class MetadataAgentConfigurationTest : public ::testing::Test {
+ public:
+  void CallParseConfiguration(MetadataAgentConfiguration& config, std::istream& stream) {
```

Comment:
Since you always pass in a `stringstream` initialized from a string, and you have to call this function anyway to get access to the private method, why not make this method take a string (well, `const std::string&`) and do the stream stuff in this method?

Reply:
Fixed all.